src/log.rs

```diff
@@ -116,19 +116,42 @@ impl Rav1dLog for Option<Rav1dLogger> {
     }
 }
 
-/// Used as a marker for [`Rav1dLogger::Stdout`].  Still a valid (i.e. safe) [`Dav1dLoggerCallback`], though.
-unsafe extern "C" fn rav1d_logger_stdout(_cookie: *mut c_void, _fmt: *const c_char, ...) {}
+mod marker {
+    use super::*;
 
-/// Used as a marker for [`Rav1dLogger::Stderr`].  Still a valid (i.e. safe) [`Dav1dLoggerCallback`], though.
-unsafe extern "C" fn rav1d_logger_stderr(_cookie: *mut c_void, _fmt: *const c_char, ...) {}
+    type Callback = extern "C" fn(cookie: *mut c_void, fmt: *const c_char);
+
+    const fn cast(callback: Callback) -> Dav1dLoggerCallback {
+        // Safety: It should always be safe to ignore variadic args.
+        // Declaring a variadic `fn` is unstable, though, which is why we avoid that.
+        unsafe { std::mem::transmute(callback) }
+    }
+
+    /// Create an empty [`Dav1dLoggerCallback`] for use as a marker `fn`
+    /// for special `fn`s stored in [`Dav1dLogger::callback`].
+    macro_rules! create {
+        ($name:ident) => {{
+            extern "C" fn $name(_cookie: *mut c_void, _fmt: *const c_char) {
+                // The `fn` needs a unique body so that
+                // multiple ones don't get optimized into the same `fn`.
+                unimplemented!(stringify!($name));
+            }
+
+            cast($name)
+        }};
+    }
```

Comment:
Don't add a macro for two uses, just duplicate the code. Same with the cast function, just write this all out explicitly. Macros are only good when they save ongoing effort and are clear, not for a one-off use like this.

Reply:
Okay, I can expand it, though keep in mind that if we add new variants to `enum Rav1dLogger` (which we anticipate to write to a `dyn io::Write` or similar once we know what common use cases are), we'll need to create more of these.

Reply:
Expanded in c310b04.